src/open_dread_rando/files/custom_scenario.lua

```diff
@@ -476,4 +490,27 @@ end
 
 function Scenario.UpdateRoomName(new_subarea)
     RoomNameGui.Update(new_subarea)
+end
+
+function Scenario.UpdateNumTanksMax(num_locs)
+    -- sets Scenario blackboard NumTanksPickedUpMax to num_locs
+    Game.LogWarn(0, "Setting " .. CurrentScenarioID .. ".NumTanksPickedUpMax = " .. num_locs)
+    Scenario.WriteToBlackboard("NumTanksPickedUpMax", "i", num_locs)
+end
+
+function Scenario.IncrementCompletion()
+    -- increment global NumTanksPickedUp
+    local global_itemcount = Scenario.GetBlackboardProp("GAME", "NumTanksPickedUp")
+    if global_itemcount == nil then -- if its the first pickup its uninitialized
+        global_itemcount = 0
+    end
+    global_itemcount = global_itemcount + 1
+    Scenario.SetBlackboardProp("GAME", "NumTanksPickedUp", "i", global_itemcount)
+
+    -- update global completion number (the "percent" which is stored as an int for reasons unknown)
+    Scenario.SetBlackboardProp("GAME", "Completion", "i", math.floor(global_itemcount * 100 / 149))
```

Comment:
semantic difference here, but I would prefer `math.floor(global_itemcount / 149) * 100` as it more clearly indicates that you're multiplying a ratio by 100 to express it as a percentage

Reply:
that would make it `0` every time lol

Reply:
oh I put the close paren in the wrong spot you get the idea tho